File: examples-sdk/swift/015-basic-mqtt-iot-gateway/main.swift
// 015-basic-mqtt-iot-gateway — Swift SDK equivalent
// Compile: vil compile --from swift --input 015-basic-mqtt-iot-gateway/main.swift --release

let server = VilServer(name: "mqtt-iot-gateway", port: 8080)
let mqtt_iot = ServiceProcess(name: "mqtt-iot")
mqtt_iot.endpoint(method: "POST", path: "/sensors/data", handler: "receive_sensor_data")
mqtt_iot.endpoint(method: "GET", path: "/sensors", handler: "list_sensors")
mqtt_iot.endpoint(method: "GET", path: "/mqtt/config", handler: "mqtt_config")
mqtt_iot.endpoint(method: "GET", path: "/mqtt/topics", handler: "mqtt_topics")
server.service(mqtt_iot)
let root = ServiceProcess(name: "root")
server.service(root)
server.compile()

File: examples-sdk/csharp/015-basic-mqtt-iot-gateway/Main.cs
// 015-basic-mqtt-iot-gateway — C# SDK equivalent
// Compile: vil compile --from csharp --input 015-basic-mqtt-iot-gateway/Main.cs --release

#r "sdk/csharp/Vil.cs"

var server = new VilServer("mqtt-iot-gateway", 8080);
var mqtt_iot = new ServiceProcess("mqtt-iot");
mqtt_iot.Endpoint("POST", "/sensors/data", "receive_sensor_data");
mqtt_iot.Endpoint("GET", "/sensors", "list_sensors");
mqtt_iot.Endpoint("GET", "/mqtt/config", "mqtt_config");
mqtt_iot.Endpoint("GET", "/mqtt/topics", "mqtt_topics");
server.Service(mqtt_iot);
var root = new ServiceProcess("root");
server.Service(root);
server.Compile();
